### PR TITLE
feat: show quest as completed after all actions have been completed

### PR DESCRIPTION
**Pull request by ribeirojose**

### CODE REVIEW

File: apps/govquests-api/rails_app/app/graphql/types/quest_type.rb [hunk 1]
```diff
@@ -7,9 +7,14 @@ class QuestType < Types::BaseObject
     field :rewards, [Types::RewardType], null: false
     field :display_data, Types::DisplayDataType, null: false
     field :actions, [Types::ActionType], null: false
+    field :user_quests, [Types::UserQuestType], null: false
 
     def actions
       object.quest_actions.order(:position).map(&:action)
     end
+
+    def user_quests
```
Comment (ribeirojose) [inline apps/govquests-api/rails_app/app/graphql/types/quest_type.rb:16]: `#could` use policy here

File: apps/govquests-api/rails_app/db/migrate/20241011130452_add_user_quest_id_to_user_quests.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class AddUserQuestIdToUserQuests < ActiveRecord::Migration[8.0]
+  def change
+    change_table :user_quests do |t|
+      t.string :user_quest_id, null: false, index: {unique: true}
```
Comment (devjoaov) [inline apps/govquests-api/rails_app/db/migrate/20241011130452_add_user_quest_id_to_user_quests.rb:4]: why not using an "id" here? 🤔

File: apps/govquests-api/rails_app/app/read_models/questing/user_quest_read_model.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 module Questing
   class UserQuestReadModel < ApplicationRecord
     self.table_name = "user_quests"
+
+    belongs_to :quest, class_name: "Questing::QuestReadModel", foreign_key: "quest_id", primary_key: "quest_id"
```
Comment (devjoaov) [inline apps/govquests-api/rails_app/app/read_models/questing/user_quest_read_model.rb:5]: 👍

**Review by devjoaov — APPROVED** [review]

File: apps/govquests-api/rails_app/db/migrate/20241011130452_add_user_quest_id_to_user_quests.rb (hunk 2 above)
Comment (ribeirojose) [reply apps/govquests-api/rails_app/db/migrate/20241011130452_add_user_quest_id_to_user_quests.rb:4]: just the convention we established to prefix all primary keys with the table entity

**Pull request merged**